src/pointer-analysis/dereference.cpp

```diff
@@ -1078,6 +1076,10 @@ void dereferencet::construct_from_array(
     unsigned int num_bytes = compute_num_bytes_to_extract(
       replaced_dyn_offset, type_byte_size_bits(type).to_uint64());
 
+    if(!num_bytes)
+      num_bytes = compute_num_bytes_to_extract(
+        offset, type_byte_size_bits(value->type).to_uint64());
```

Comment:
Why is it OK to pass a possibly non-constant `offset` here in contrast to the `replaced_dyn_offset` above, which should be constant (after simplification) in more cases that `offset`? I fail to see what dynamic offsets have to do with FAMs.

Reply:
Nice catch! This was some leftover of the previous approach where we would not update the array size at the frontend. I will remove it.